docs/dev/topology-over-raft.md

```diff
@@ -114,11 +118,28 @@ that there are no tablet transitions in the system.
 Tablets are migrated in parallel and independently.
 
 There is a variant of tablet migration track called tablet draining track, which is invoked
-as a step of certain topology operations (e.g. decommission, removenode, replace). Its goal is to readjust tablet replicas
+as a step of certain topology operations (e.g. decommission, removenode). Its goal is to readjust tablet replicas
 so that a given topology change can proceed. For example, when decommissioning a node, we
 need to migrate tablet replicas away from the node being decommissioned.
 Tablet draining happens before making changes to vnode-based replication.
 
+## Node replace with tablets
+
+Tablet replicas on the replaced node are rebuilt after the replacing node is already in the normal state and
+the replaced node is in the left state.
+
+Until old replicas are rebuilt, the availability in the cluster is reduced. If another node becomes unavailable, we
+may have two unavailable replicas for some tablets. Admin needs to know that and not start rolling restart for example.
+To avoid surprises, the replaced node waits on boot for tablet replicas to finish rebuilding
+so that admin sees the replace as finished after availability was restored.
+
+### Impact on repair
+
+When tablet is rebuilt in the background after replace, its primary replica may be on the node which is no
+longer in topology. This means that running repair -pr on all nodes will not repair such a tablet, but it's fine because
```

Comment:
Don't we have the same problem with vnodes?
When ignoring dead nodes for repair, maybe the primary replica should be the next one in replication strategy order.

Cc @asias

Reply:
If the dead node is a primary replica node, we are missing one node in the cluster until the replace is finished. So users will not expect the repair will repair all ranges. 

I do not think it is a good idea to chance the meaning of the primary replica to the next one and have a special handling. One can just run repair after the replace is done. Also, we use repair on the replacing node anyway.

Reply:
I see your point.  Sigh, current repair api is too complicated and hard to use when the cluster isn't in a perfect shape (when you probably need repair the most).

I look forward for a simple, cluster-wide native-repair api that will simplify the admin's life and repair everything without having to reverse engineer the replication_map.

It may still have options like controlling if repair would be local in each data center or should it repair across data centers (or an automatic mode that will use local-dc repair only when it's safe and there are enough replicas in the local dc to form a quorum for each tablet), but it won't require the user to intimately understand the replication map, token ranges, primary ownership, etc.